aiida_lsmo/workchains/multicomp_ads_des.py

```diff
@@ -0,0 +1,351 @@
+                              dynamic=True,
+                              help='Generated block pocket files.')
+
+    def setup(self):
+        """Initialize parameters"""
+        self.ctx.sim_in_box = "structure" not in self.inputs.keys()
+        self.ctx.parameters = aiida_dict_merge(Dict(dict=PARAMETERS_DEFAULT), self.inputs.parameters)
+        self.ctx.components = get_components_dict(self.inputs.conditions, self.ctx.parameters)
+        self.ctx.ff_params = get_ff_parameters(self.ctx.components, self.ctx.parameters)
+
+    def should_run_zeopp(self):
+        """Return if it should run zeopp calculation."""
+        return not self.ctx.sim_in_box and self.ctx.parameters['zeopp_probe_scaling'] > 0.0
+
+    def run_zeopp(self):
+        """It performs the full zeopp calculation for all components."""
+        zeopp_inputs = self.exposed_inputs(ZeoppCalculation, 'zeopp')
+        zeopp_inputs.update({'structure': self.inputs.structure, 'atomic_radii': get_atomic_radii(self.ctx.parameters)})
+        for key, value in self.ctx.components.get_dict().items():
+            comp = value['name']
+            zeopp_label = "Zeopp_{}".format(comp)
+            zeopp_inputs['metadata']['label'] = zeopp_label
+            zeopp_inputs['metadata']['call_link_label'] = "run_{}".format(zeopp_label)
+            zeopp_inputs['parameters'] = ZeoppParameters(dict=self.ctx.components[key]['zeopp'])
+            running = self.submit(ZeoppCalculation, **zeopp_inputs)
+            self.report(f"Running zeo++ block calculation<{running.id}> for {comp}")
+            self.to_context(**{zeopp_label: running})
+
+    def inspect_zeopp_calc(self):
+        """Asserts whether all widom calculations are finished ok. If so, manage zeopp results."""
+        for value in self.ctx.components.get_dict().values():
+            assert self.ctx["Zeopp_{}".format(value['name'])].is_finished_ok
+
+        for value in self.ctx.components.get_dict().values():
+            comp = value['name']
+            zeopp_label = f"Zeopp_{comp}"
+            if self.ctx[zeopp_label].outputs.output_parameters['Number_of_blocking_spheres'] > 0:
+                self.out(f"block_files.{comp}_block_file", self.ctx[zeopp_label].outputs.block)
+
+    def _get_gcmc_inputs_adsorption(self):
+        """Generate Raspa input parameters from scratch, for a multicomponent GCMC calculation."""
+        self.ctx.raspa_inputs = self.exposed_inputs(RaspaBaseWorkChain, 'raspa_base')
+        self.ctx.raspa_inputs['raspa']['file'] = FFBuilder(self.ctx.ff_params)
+        self.ctx.raspa_inputs["raspa"]["block_pocket"] = {}
+        self.ctx.raspa_param = {
+            "GeneralSettings": {
+                "SimulationType": "MonteCarlo",
+                "NumberOfInitializationCycles": self.ctx.parameters['raspa_gcmc_init_cycles'],
+                "NumberOfCycles": self.ctx.parameters['raspa_gcmc_prod_cycles'],
+                "PrintPropertiesEvery": int(1e10),
```

Comment:
is this your way of saying "never"?

perhaps better to have a special value for this (e.g. -1)

Reply:
Better not to introduce other convention: -1 or 0 would make Raspa crash